# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/js.msg
# Commit: b1117df3b91d
# Full Hash: b1117df3b91d10d68d10b399637cf9ac295057e6
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2019-11-01 16:13:11
# Regressor Bug: 1587096
# File Overlap Count: 1
# Description:
#   Bug 1587096 - Part 3: Implmement FinalizationGroup r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49946
# ==============================================================================

diff -r cefc0474e13c -r b1117df3b91d js/src/js.msg
--- a/js/src/js.msg	Fri Nov 01 10:37:40 2019 +0000
+++ b/js/src/js.msg	Fri Nov 01 10:37:47 2019 +0000
@@ -714,3 +714,11 @@
 
 // BinAST
 MSG_DEF(JSMSG_BINAST,                                    1, JSEXN_SYNTAXERR, "BinAST Parsing Error: {0}")
+
+// FinalizationGroup
+MSG_DEF(JSMSG_NOT_A_FINALIZATION_GROUP,                  1, JSEXN_TYPEERR, "{0} is not a FinalizationGroup")
+MSG_DEF(JSMSG_NOT_A_FINALIZATION_ITERATOR,               1, JSEXN_TYPEERR, "{0} is not a FinalizationGroupCleanupIterator")
+MSG_DEF(JSMSG_BAD_HOLDINGS,                              0, JSEXN_TYPEERR, "The holdings passed to FinalizationGroup.register must not be the same as its target object")
+MSG_DEF(JSMSG_BAD_UNREGISTER_TOKEN,                      1, JSEXN_TYPEERR, "Invalid unregister token passed to {0}")
+MSG_DEF(JSMSG_STALE_FINALIZATION_GROUP_ITERATOR,         0, JSEXN_TYPEERR, "Can't use stale finalization group iterator")
+MSG_DEF(JSMSG_BAD_CLEANUP_STATE,                         0, JSEXN_TYPEERR, "Can't call FinalizeGroup.cleanupSome while cleanup is in progress")